frame/codec.go

```diff
@@ -0,0 +1,136 @@
+package frame
+
+import (
+	"fmt"
+	"github.com/datastax/go-cassandra-native-protocol/message"
+	"github.com/datastax/go-cassandra-native-protocol/primitive"
+	"io"
+)
+
+type Encoder interface {
+
+	// EncodeFrame encodes the entire frame, compressing the body if needed.
+	EncodeFrame(frame *Frame, dest io.Writer) error
+}
+
+type RawEncoder interface {
+
+	// EncodeRawFrame encodes the given RawFrame.
+	EncodeRawFrame(frame *RawFrame, dest io.Writer) error
+
+	// EncodeHeader encodes the given frame Header. This is a partial operation; after calling this method, one must
+	// call EncodeBody to encode the to fully encode the entire frame.
+	EncodeHeader(header *Header, dest io.Writer) error
+
+	// EncodeBody encodes the given frame Body. The body will be compressed depending on whether the Compressed flag is
+	// set in the given Header. This is a partial operation; it is illegal to call this method before calling
+	// EncodeHeader.
+	EncodeBody(header *Header, body *Body, dest io.Writer) error
+}
```

Comment:
I understand the reason behind splitting the `Encoder` and `RawEncoder` interfaces but is there enough value to be gained from this to justify the additional hurdle of creating two encoder instances (and 2 decoders) if the user wants both?

Reply:
RawCodec inherits from Codec so if a user wants both they could simply go with RawCodec.

Reply:
oh I missed that 👍